src/org/labkey/remoteapi/query/InsertExternalSchemaCommand.java

```diff
@@ -0,0 +1,166 @@
+package org.labkey.remoteapi.query;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.message.BasicNameValuePair;
+import org.junit.Assert;
+import org.labkey.remoteapi.CommandException;
+import org.labkey.remoteapi.CommandResponse;
+import org.labkey.remoteapi.Connection;
+import org.labkey.remoteapi.PostCommand;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+public class InsertExternalSchemaCommand extends PostCommand<CommandResponse>
+{
+    private final Params _params;
+
+    public InsertExternalSchemaCommand(Params params)
+    {
+        super("query", "insertExternalSchema");
+        _params = params;
+    }
+
+    @Override
+    protected HttpUriRequest createRequest(URI uri)
+    {
+        HttpPost request = (HttpPost) super.createRequest(uri);
+
+        try
+        {
+            request.setEntity(_params.toFormEntity());
+        }
+        catch (UnsupportedEncodingException e)
+        {
+            throw new RuntimeException(e);
+        }
+
+        return request;
+    }
+
+    @Override
+    public CommandResponse execute(Connection connection, String folderPath) throws IOException, CommandException
+    {
+        CommandResponse response = super.execute(connection, folderPath);
+        validateTables(connection, folderPath);
+        return response;
+    }
+
+    private void validateTables(Connection connection, String folderPath) throws IOException, CommandException
+    {
+        GetQueriesCommand command = new GetQueriesCommand(_params.userSchemaName);
+        GetQueriesResponse response = command.execute(connection, folderPath);
+        try
+        {
+            Assert.assertEquals(String.format("Wrong tables published to %s schema '%s'.", _params.schemaType(), _params.userSchemaName),
+                    new HashSet<>(_params.tables), new HashSet<>(response.getQueryNames()));
+        }
+        catch (AssertionError e)
+        {
+            throw new CommandException(e.getMessage());
+        }
```

Comment:
Just curious...
I'm assuming that you are catching this and throwing it as a CommandException for consistency (everything throws CommandException), is that why?